hypershift-operator/controllers/nodepool/metrics/metrics.go

```diff
@@ -230,41 +308,15 @@ func (c *nodePoolsMetricsCollector) retrieveVCpusDetailsPerNode(nodePool *hyperv
 	if vCpusCountPerNode, isCached := c.ec2InstanceTypeToVCpusCount[ec2InstanceType]; isCached {
 		return vCpusDetail{vCpusCount: vCpusCountPerNode}
 	}
-	awsInput := ec2.DescribeInstanceTypesInput{InstanceTypes: []*string{&ec2InstanceType}}
-	unresolvedErrorReason := ""
-
-	if awsOuput, err := c.ec2Client.DescribeInstanceTypes(&awsInput); awsOuput != nil && err == nil {
-		if len(awsOuput.InstanceTypes) == 1 {
-			ec2InstanceTypeInfo := awsOuput.InstanceTypes[0]
-
-			if ec2InstanceTypeInfo != nil {
-				instanceTypeInInfo := ec2InstanceTypeInfo.InstanceType
-				cpuInfo := ec2InstanceTypeInfo.VCpuInfo
-
-				if instanceTypeInInfo != nil && *instanceTypeInInfo == ec2InstanceType && cpuInfo != nil {
-					vCpusCountPtr := cpuInfo.DefaultVCpus
-
-					if vCpusCountPtr != nil {
-						vCpusCount := int32(*vCpusCountPtr)
-
-						c.ec2InstanceTypeToVCpusCount[ec2InstanceType] = vCpusCount
-
-						return vCpusDetail{vCpusCount: vCpusCount}
-					}
-				}
-			}
-		}
 
-		unresolvedErrorReason = "unexpected AWS output"
-		ctrllog.Log.Error(errors.New(unresolvedErrorReason), "unexpected output for EC2 verb 'describe-instance-types' while querying the following EC2 instance type: "+ec2InstanceType)
-	} else {
-		unresolvedErrorReason = "failed to call AWS"
-		ctrllog.Log.Error(err, "failed to call AWS to resolve the number of vCpus per node for the following EC2 instance type: "+ec2InstanceType)
+	vCpusCount, errorReason := extractCPUFromInstanceTypeName(ec2InstanceType, c.ec2Client, c.pricingClient)
+	if errorReason != noErroReason {
+		(*ec2InstanceTypeToResolutionErrorReason)[ec2InstanceType] = errorReason
+		return vCpusDetail{vCpusCount: -1, vCpusCountErrorReason: errorReason}
 	}
-
-	(*ec2InstanceTypeToResolutionErrorReason)[ec2InstanceType] = unresolvedErrorReason
-
-	return vCpusDetail{vCpusCount: -1, vCpusCountErrorReason: unresolvedErrorReason}
+	// in case of no error we cache the value
+	c.ec2InstanceTypeToVCpusCount[ec2InstanceType] = int32(vCpusCount)
```

Comment:
This entry does not expire. This may not be okay if the `vCpusCount` is read from some config map which may change (values read from AWS are less subject to change).

Entries in this `ec2InstanceTypeToVCpusCount` cache should maybe have a counter telling how much times the value can be read from the cache before expiring and be removed from the cache. WDYT?

Reply:
So I'm lost here. What if the configMap changes? Should we restart the pod?